snapcraft/internal/repo/__init__.py

```diff
@@ -34,11 +34,19 @@ def check_for_command(command):
 
 
 def get_pkg_name_parts(pkg_name):
```

Comment:
This is getting complex, and looks error-prone. I can imagine myself asking "what's the second arg... version or arch?" and "the arch doesn't include the colon, right?"

I also see us using `name + arch` a lot below. I suggest making this a super simple class with a few helper methods to do that type of thing for you, and return an instance of that class here.

Reply:
I dropped those changes from this PR, after moving to using the error message to render instructions needed to setup the host.